tests/unit/useOnyxTest.ts

```diff
@@ -276,6 +276,8 @@ describe('useOnyx', () => {
             expect(result.current[1].status).toEqual('loaded');
 
             selector = (entry: OnyxEntry<{id: string; name: string}>) => `id - ${entry?.id}, name - ${entry?.name} - selector changed`;
+            // In a react app we expect the selector ref to change during a rerender (see selectorRef/useLiveRef)
+            rerender(undefined);
 
             await act(async () => Onyx.merge(ONYXKEYS.TEST_KEY, {id: 'changed_id', name: 'changed_name'}));
```

Comment:
```suggestion
            await act(async () => waitForPromisesToResolve());

            expect(result.current[0]).toEqual('id - test_id, name - test_name - selector changed');
```

I think we can remove the Onyx.merge since the rerender() is enough to prove the selector changed

Reply:
ah good idea, let me check!

Reply:
We don't even need waitForPromisesToResolve() 😊 looks good?